substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jni/functions/JNIFunctions.java

```diff
@@ -872,6 +900,9 @@ interface NewObjectWithObjectArrayArgFunctionPointer extends CFunctionPointer {
     static int ThrowNew(JNIEnvironment env, JNIObjectHandle clazzHandle, CCharPointer message) throws Throwable {
         Class<?> clazz = JNIObjectHandles.getObject(clazzHandle);
         JNIMethodId ctor = Support.getMethodID(clazz, "<init>", "(Ljava/lang/String;)V", false);
+        if (ctor.isNull()) {
+            return JNIErrors.JNI_EINVAL();
+        }
```

Comment:
`Support.getMethodID` already throws when it cannot find the constructor, and so does the HotSpot implementation of `ThrowNew`.

Reply:
That's true, but that's not what I understand from [the JNI specifications](https://docs.oracle.com/en/java/javase/11/docs/specs/jni/functions.html#thrownew). `ThrowNew()` is supposed to throw the "newly constructed `java.lang.Throwable` object", and return a negative value of failure to do that (in this case, `JNI_EINVAL`).

Reply:
This is a generic phrase in the JNI specification. In HotSpot's implementation which we consider the reference for implementation details, I can't see a code path in which it wouldn't return `JNI_OK`.

Reply:
I don't like that, but okay.